paimon-core/src/main/java/org/apache/paimon/partition/actions/PartitionMarkDoneAction.java

```
@@ -29,36 +29,62 @@
+                                case SUCCESS_FILE:
                                     return new SuccessFileMarkDoneAction(
                                             fileStoreTable.fileIO(), fileStoreTable.location());
-                                case "done-partition":
+                                case DONE_PARTITION:
                                     return new AddDonePartitionAction(
                                             createMetastoreClient(fileStoreTable, options));
-                                case "mark-event":
+                                case MARK_EVENT:
                                     return new MarkPartitionDoneEventAction(
                                             createMetastoreClient(fileStoreTable, options));
+                                case CUSTOM:
+                                    return getCustomMarkDoneAction(cl, options);
                                 default:
                                     throw new UnsupportedOperationException(action);
                             }
                         })
                 .collect(Collectors.toList());
     }
 
+    static PartitionMarkDoneAction getCustomMarkDoneAction(ClassLoader cl, CoreOptions options) {
+        if (options.partitionMarkDoneCustomClass() == null) {
+            throw new IllegalArgumentException(
+                    String.format(
+                            "You need to set [%s] when you add [%s] mark done action in your property [%s].",
+                            PARTITION_MARK_DONE_CUSTOM_CLASS.key(),
+                            CUSTOM,
+                            PARTITION_MARK_DONE_ACTION.key()));
+        }
+        String customClass = options.partitionMarkDoneCustomClass();
+        try {
+            return (PartitionMarkDoneAction) cl.loadClass(customClass).newInstance();
+        } catch (ClassNotFoundException | IllegalAccessException | InstantiationException e) {
```

Can we catch all exceptions here?